### DISCUSSION TITLE
Is it possible to get multiple argo instances on a same openshift cluster?

### DISCUSSION THREAD

**Discussion by sylock** [argoproj/argo-workflows#4843] [Q&A] [answered]
I would like to get some different instances of Argo on the same openshift cluster. The idea would be to have a development version where everybody can try and break things. And a production one with production workflows.

I tried to do it but I get this error : "Forbidden: workflows.argoproj.io is forbidden: User "system:serviceaccount:argo-wf-development:argo-server" cannot list resource "workflows" in API group "argoproj.io" at the cluster scope"

I tried to have the development in the "namespaced" mode and the production in the "cluster" mode. But it doesn't change anything since the CRD are the same.

What do you think? Has anybody already tried?

**Comment by alexec** [comment, marked as answer]
I don't think anyone would recommend have both production and development workloads in a single cluster. You do not have isolation so at some point your development workloads will impact your production workflows, e.g. by accidentally consuming all resources in the cluster. This risk will grow with time, so plan for a production outage as a consequence.

More generally, you can run multiple workflow controllers in a cluster using one of these options:

* One namespace-scoped install per namespace. 
* Multiple cluster-scoped installs that use instance ID configuration (see the docs).

**Reply by smals-nim** [reply]
Hi Alex,

Thank you very much. It is exactly the information I needed to take my decision.
Thanks for responding so quickly 👍 

Have a nice day,

**Reply by AnjanaAK** [reply]
> I don't think anyone would recommend have both production and development workloads in a single cluster. You do not have isolation so at some point your development workloads will impact your production workflows, e.g. by accidentally consuming all resources in the cluster. This risk will grow with time, so plan for a production outage as a consequence.
> 
> More generally, you can run multiple workflow controllers in a cluster using one of these options:
> 
> * One namespace-scoped install per namespace.
> * Multiple cluster-scoped installs that use instance ID configuration (see the docs).

Hi Alex, @alexec 

We already have a cluster-wide argo workflows installation.
Need one more of another version in the same cluster.
So, will a new namespace-scoped install in a separate namespace work with that?

And could you please give some more information about what exactly meant by **isolation** ? 
Is it just the problem of shared resource consumption?

**Comment by sylock** [comment]
Yes you can close the ticket. Thanks.
